db/primary_migrate/20220129181752_create_sign_in_restrictions.rb

```diff
@@ -0,0 +1,17 @@
+class CreateSignInRestrictions < ActiveRecord::Migration[6.1]
+  def change
+    create_table :sign_in_restrictions do |t|
+      t.integer :user_id, null: false
+      t.string :service_provider
+
```

Comment:
maybe for the future we can add a followup for with a reason (internal only, just for record keeping)
```suggestion
      t.string :reason
```